### PR TITLE
Correct motor positions database

### PR DESCRIPTION
**Pull request by keara-soloway**
Correct database setup & interaction codes for the motor positions database

### CODE REVIEW

**Comment by keara-soloway** [comment]
@vkuznet I have replaced DID with ScanID as the unique key for lookup in the motor positions db. ScanId is a uint64 taken from the record's StartTime (which is a float64). Do you see any other issues that should be resolved before merging?

**Comment by vkuznet** [comment]
@keara-soloway , I think we can merge and iterate further. What I'm interested is to see how look up will be done. For instance:
1. user provides did
2. user provides did and motor position/name 
3. user provides motor position and/or name but no did

How in each case we can look up the data from this service. For example, for case first we need to look-up data from MongoDB first, then take sid's and get all records from relational DB, while for third case it will be reverse, i.e. we get all records with sids and then query MongoDB with such list. Can you check that these case will be covered.

**Comment by keara-soloway** [comment]
All three cases you mentioned are addressed in SearchHandler, but cases 2. and 3. are covered imperfectly. The outstanding issues:
1. One cannot search for scans in which a motor with a particular name is simply present. Any query with a motor name currently _must_ include a position, too.
2. There is no way for one to "or" criteria for did (or any other values kept in the mongodb) with criteria for motor name & position. If one's query includes criteria for did as well as motor name / position, all the records in the results will always satisfy _both_ criteria. 
3. One cannot search for scans based on criteria for more than 1 motor.
4. One may only query on an _exact match_ for a motor position. 

Items 3. and 4. are caused by limitations of the keys I chose to allow for querying on motor positions. I will commit my current json file of db query keys, knowing that it (& SearchHandler) will need to be updated to fix these shortcomings.

I will merge this PR for now & continue to iterate elsewhere.

**Comment by vkuznet** [comment]
Keara, let me comment on your cases:
1. I don't see why someone must provide position to the motor, you just need to relax the WHERE clause, e.g. `select * from TABLE where motor_name = 'bla'`
2. if you want to support queries with multiple did's it should be possible if did in motorsdb, e.g. here is how you will construct the query: `select * from table where did=/abc or did=/xyz`
3. seems like a previous case, e.g. `select * from table where motor_name=abc or montor_name=xyz`
4. you may use `LIKE` or `>` or `<` operators in queries, like:
```
# use % as wildcard
select * from TABLE where montor_name like %ab%

# use less/greater operator for numerical values
select * from TABLE where position > 1 and position < 2
```

Bottom line, the SQL query syntax is very rich and allows to compose complex queries. It seems that did still need to be presented motorsdb tables as this may be easier to use in searches in stead of using sids in mongodb records. What you need is to lay down both possibilities and see how it will affect queries.

**Comment by keara-soloway** [comment]
Sorry, I may not have been clear enough. The SQL statements for those cases are already there -- one can use a MotorsDbQuery struct to fill out the template  in static/query_motorsdb.sql, and all cases should be accounted for. However, the current SearchHandler will only parse two values from the user's query ("motor" and "position"), and these two values alone don't allow for a translation into any arbitrary MotorsDbQuery struct.

**Comment by keara-soloway** [comment]
Ah, thanks, I see now! Then I'll try writing the translation between user query & MotorsDbQuery with the allowed keys as-is.
